src/function.calls.spec.ts

```diff
@@ -155,4 +155,76 @@ describe("Function calls detector Agent Tests", () => {
     const findings: Finding[] = await handleTransaction(txEvent);
     expect(findings).toStrictEqual([findingGenerator({ to, from, input })]);
   });
+
+  it("Should be able to handle additional filter function passed inside functionCallGenerator", async () => {
+    const abiItem = {
+      name: "myMethod",
+      type: "function",
+      inputs: [
+        {
+          type: "uint256",
+          name: "myNumber",
+        },
+        {
+          type: "uint256",
+          name: "myNumber",
+        },
+      ],
+    } as AbiItem;
+
+    const filterFn = (value: TraceAction): boolean => {
+      const input = value.input;
+      const decode = abiDecode([abiItem], input) as any;
+      if (decode.params[0].value === "1000" && decode.params[1].value === "100") return false;
+      else return true;
+    };
+
+    const input = abi.encodeFunctionCall(abiItem, ["1000", "100"]);
+    const falseValue = filterFn({ input } as any);
+
+    const inputForFailingCase = abi.encodeFunctionCall(abiItem, ["100", "100"]);
+    // dont emit a warning if return value is true for our case
+    const trueValue = filterFn({ input: inputForFailingCase } as any);
+
+    expect(trueValue).toBe(true);
+    expect(falseValue).toBe(false);
```

Comment:
this lines are for testing the `filterFn` function and that is not needed because the function is defined above inside the test